src/twisted/runner/test/test_procmon.py

```diff
@@ -279,6 +284,16 @@ def test_startProcessUnknownKeyError(self):
         """
         self.assertRaises(KeyError, self.pm.startProcess, "foo")
 
+    def test_startProcessSpawnException(self):
+        """
+        L{IReactorProcess.spawnProcess} might raise C{OSError}, ensure it
+        is caught and process is restarted
+        """
+        self.reactor._spawn_resource_unavailable = True
```

Comment:
As @adiroiban pointed out, this should really be a public API on the control surface of the reactor.  Given that the class is just within this one test file, it should be fine to just add it as a method.